fabric-rendering-v1/src/main/java/net/fabricmc/fabric/api/client/render/v1/BlockEntityRendererRegistry.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.render.v1;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import net.minecraft.block.entity.BlockEntity;
+import net.minecraft.client.render.block.entity.BlockEntityRenderDispatcher;
+import net.minecraft.client.render.block.entity.BlockEntityRenderer;
+
+/**
+ * Helper class for registering BlockEntityRenderers.
+ */
+public class BlockEntityRendererRegistry {
+	public static final BlockEntityRendererRegistry INSTANCE = new BlockEntityRendererRegistry();
+	private Map<Class<? extends BlockEntity>, BlockEntityRenderer<? extends BlockEntity>> renderers = null;
+	private Map<Class<? extends BlockEntity>, BlockEntityRenderer<? extends BlockEntity>> renderersTmp = new HashMap<>();
+
+	private BlockEntityRendererRegistry() {
+	}
+
+	public void initialize(BlockEntityRenderDispatcher instance, Map<Class<? extends BlockEntity>, BlockEntityRenderer<? extends BlockEntity>> map) {
```

Comment:
Don't merge the impl class and the api class, see https://github.com/FabricMC/fabric/blob/d54afe87685024ae2d87b5a0d2684d4824f9701c/fabric-renderer-registries-v1/src/main/java/net/fabricmc/fabric/api/client/rendereregistry/v1/BlockEntityRendererRegistry.java#L31

Reply:
1.14 api didn't have that

Reply:
yeah that's improved in the newer versions

Reply:
done